Marlin/Configuration.h

```diff
@@ -1144,8 +1144,14 @@
 //#define Z_SAFE_HOMING
 
 #if ENABLED(Z_SAFE_HOMING)
-  #define Z_SAFE_HOMING_X_POINT ((X_BED_SIZE) / 2)    // X point for Z homing when homing all axes (G28).
-  #define Z_SAFE_HOMING_Y_POINT ((Y_BED_SIZE) / 2)    // Y point for Z homing when homing all axes (G28).
+  #if ENABLED(AUTO_BED_LEVELING_UBL)
+  //Try to home on a grid point to better allow checking mesh center
+    #define Z_SAFE_HOMING_X_POINT ((GRID_MAX_POINTS_X/2)*(X_BED_SIZE-2*MESH_INSET)/(GRID_MAX_POINTS_X-1)+MESH_INSET) // X point for Z homing when homing all axes (G28).
+    #define Z_SAFE_HOMING_Y_POINT ((GRID_MAX_POINTS_Y/2)*(Y_BED_SIZE-2*MESH_INSET)/(GRID_MAX_POINTS_Y-1)+MESH_INSET) // Y point for Z homing when homing all axes (G28).
```

Comment:
The point of these settings is to allow users to override a default. If you think this is a more sensible default (closer to the center of the bed) when UBL is enabled, then this code should be in `Conditionals_LCD.h` as the fallback when there are no `Z_SAFE_HOMING_*_POINT` values set. There's no way we're going to propagate this to every configuration file as the "elegant" default for UBL.

Reply:
Ok, will do, thanks. Those expressions cause homing to occur on a grid point as close as close as possible to the center of the bed. It will only be exactly at the center of the bed if GRID_MAX_POINTS_X is odd (current default is 10). 
The rationale for this: 
I recently tried turned on UBL leveling for the first time. I was expecting that after a 
`"G28\nG29 P1\nG28\nG29 T"` sequence, the "current value" in the table would be close to 0 because my z probe is pretty reproducible. When the value was an order of magnitude greater than I was expecting I spent quite a while rechecking my system etc. Eventually I realized the discrepancy was due to two things, first the center of the bed homing position was not on a grid point but halfway between grid points, and second, the grid point marked as the "current position" was not taking the horizontal z probe offsets into account.
My thinking was that these changes could save someone from wasting a similar amount of time in the future, although putting it in `Conditionals_post.h` means no newbie will get the fix because the  `Z_SAFE_HOMING_*_POINT`  lines are not commented out by default in `Configuration.h`. 
An alternative would be to default GRID_MAX_POINTS_X to 9 or 11 rather than 10.

Reply:
> the center of the bed homing position was not on a grid point

It is seldom the case that the homing position is exactly on a grid point. If that matters, it would be better to fix the issue that causes it to matter rather than to force users to make their homing position align exactly to a grid point.

> An alternative would be to default GRID_MAX_POINTS_X to 9 or 11 rather than 10.

Whatever works best. I'm trying to avoid adding too many conditions and repetitions of the same options in the configuration files, as this complicates developing a configuration utility and generally adds more noise the the configs.

Reply:
> It is seldom the case that the homing position is exactly on a grid point. If that matters, it would be better to fix the issue that causes it to matter rather than to force users to make their homing position align exactly to a grid point.
> 
I would imagine most users use the 'calculated' default homing position rather than setting it explicitly, especially if SAFE_HOMING is defined, as is recommended when using UBL.  I.e. I doubt many new users would adjust the SAFE_HOMING position.
The homing position only matters because it is used as the z origin when measuring the grid and therefore one would expect the z offset measured there to be zero to within the homing error standard deviation.
However, with the current defaults, you cannot see this in the G29 T table, which, as a newbie to UBL, confused me because I thought the UBL scheme, or perhaps my probe, was not working well.

So this is simply about improving the experience for new users by having better defaults rather than forcing anything. The issue that causes this to matter is only that the current defaults can make things confusing for new users of UBL. Configuration.h already advises "If using a Probe for Z Homing, enable Z_SAFE_HOMING also!", so why does it not make the home position a nice one for the G29 T report?
 
> > An alternative would be to default GRID_MAX_POINTS_X to 9 or 11 rather than 10.
> 
> Whatever works best. I'm trying to avoid adding too many conditions and repetitions of the same options in the configuration files, as this complicates developing a configuration utility and generally adds more noise the the configs.

The two options I can see for improving the defaults that require modification of configuration.h would be:

1. comment out the lines `  
#define Z_SAFE_HOMING_X_POINT ((X_BED_SIZE) / 2)    // X point for Z homing when homing all axes (G28).
  #define Z_SAFE_HOMING_Y_POINT ((Y_BED_SIZE) / 2)    // Y point for Z homing when homing all axes (G28).
` so the new defaults I added in conditionals_post.h are actually used by default, or

2. change the line `#define GRID_MAX_POINTS_X 10` to `#define GRID_MAX_POINTS_X 9` or  `#define GRID_MAX_POINTS_X 11`.

I have a slight preference for option 1 because it works whether or not GRID_MAX_POINTS_* is odd.